microsoft-edge/progressive-web-apps-chromium/how-to/push.md

```diff
@@ -6,155 +6,141 @@ ms.author: msedgedevrel
 
-In your `pwabuilder-sw-register.js` file, append the following code:
+The following code snippet shows how to subscribe to push notifications in your PWA:
 
 ```javascript
-// Ask the user for permission to send push notifications.
-navigator.serviceWorker.ready
-    .then(function (registration) {
-        // Check if the user has an existing subscription
-        return registration.pushManager.getSubscription()
-            .then(function (subscription) {
-                if (subscription) {
-                    return subscription;
-                }
-
-                const vapidPublicKey = "PASTE YOUR PUBLIC VAPID KEY HERE";
-                return registration.pushManager.subscribe({
-                    userVisibleOnly: true,
-                    applicationServerKey: urlBase64ToUint8Array(vapidPublicKey)
-                });
-            });
+async function subscribeToPushMessages() {
+  const serviceWorkerRegistration = await navigator.serviceWorker.ready;
+
+  // Check if the user has an existing subscription
+  let pushSubscription = serviceWorkerRegistration.pushManager.getSubscription();
+  if (pushSubscription) {
+    // The user is already subscribed to push notifications
+    return;
+  }
+
+  try {
+    // Subscribe the user to push notifications
+    pushSubscription = await serviceWorkerRegistration.pushManager.subscribe({
+      userVisibleOnly: true,
+      applicationServerKey: urlBase64ToUint8Array("YOUR PUBLIC VAPID KEY HERE")
     });
+  } catch (err) {
+    // The subscription wasn't successful.
+    console.log("Error", err);
+  }
+}
 
 // Utility function for browser interoperability
 function urlBase64ToUint8Array(base64String) {
-    var padding = '='.repeat((4 - base64String.length % 4) % 4);
-    var base64 = (base64String + padding)
-        .replace(/\-/g, '+')
-        .replace(/_/g, '/');
-
-    var rawData = window.atob(base64);
-    var outputArray = new Uint8Array(rawData.length);
-
-    for (var i = 0; i < rawData.length; ++i) {
-        outputArray[i] = rawData.charCodeAt(i);
-    }
-    return outputArray;
+  var padding = '='.repeat((4 - base64String.length % 4) % 4);
+  var base64 = (base64String + padding)
+      .replace(/\-/g, '+')
+      .replace(/_/g, '/');
+  
+  var rawData = window.atob(base64);
+  var outputArray = new Uint8Array(rawData.length);
+  
+  for (var i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
 }
 ```
 
-See also [PushManager](https://developer.mozilla.org/docs/Web/API/PushManager) and [Web-Push](https://www.npmjs.com/package/web-push#usage).
+The VAPID key mentioned in the previous code snippet is a public key that is used to identify the server that sends the push messages and encrypt the push message payload.  See [Step 3 - Send push messages from your server](#step-3---send-push-messages-from-your-server) for more information about VAPID keys.
```

Comment:
```suggestion
The VAPID key that's mentioned in the previous code snippet is a public key that's used to identify the server that sends the push messages and encrypt the push message payload.  See [Step 3 - Send push messages from your server](#step-3---send-push-messages-from-your-server) for more information about VAPID keys.
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which